sample-app/src/service-worker.ts

```diff
@@ -0,0 +1,71 @@
+/// <reference lib="webworker" />
+import { clientsClaim } from 'workbox-core';
+import { precacheAndRoute } from 'workbox-precaching';
+
+declare const self: ServiceWorkerGlobalScope;
```

Comment:
Reading the ServiceWorker docs, I came across this statement: "After the activation step, the service worker will control all pages ... though the page that registered the service worker ... won't be controlled until it's loaded again." Do we not care about this because `registerServiceWorker` happens outside the scope of `App`?

Reply:
Yes! I thought using clientsClaim (i.e. `self.clients.claim()`) would take care of this, but that isn't exactly the case. To make this a more seamless experience I added some additional logic based on [this](https://github.com/cra-template/pwa/blob/25cb9f06938c671507684a8438c347d3c8bc8b24/packages/cra-template-pwa-typescript/template/src/serviceWorkerRegistration.ts)